app-cpp/app355.cpp

```diff
@@ -112,6 +112,7 @@ class SatOrbit
 private:
     std::size_t mNumThreads{0};
     OrbitalDataVector mOrbitalVector{};
+    std::mutex mOrbitMutex{};
```

Comment:
Q: what is the scope of `mOrbitMutex`?
meaning: Is it meant to protect the entire `SatOrbit` instance (and all data members)?
or, is it meant to only protect synchronized access to `mOrbitalVector`?

if "entire instance", then it is fine where it is (tho you should add a comment stating this)
if it is "only `mOrbitalVector`", then wrap the mutex and vector in a `std::tuple` to make it explicit.